manifests/repository.pp

```diff
@@ -1,14 +1,41 @@
 # == Class: bareos::repository
-# This class manages the bareos repository
-# Parameters should be configured in the upper class `::bareos`.
 #
-# This class will be automatically included when a resource is defined.
-# It is not intended to be used directly by external resources like node definitions or other modules.
+# @summary
+#   This class manages the bareos repository
+#   Parameters should be configured in the upper class `::bareos`.
+#
+#   This class will be automatically included when a resource is defined.
+#   It is not intended to be used directly by external resources like node definitions or other modules.
+#
+# @param release
+#   The major bareos release version which should be used
+# @param gpg_key_fingerprint
+#   The GPG fingerprint of the repos key
+# @param subscription
+#   Activate the (paid) subscription repo. Otherwise the opensource repos will be selected
+# @param username
+#   The major bareos release version which should be used
+# @param password
+#   The major bareos release version which should be used
+#
 class bareos::repository (
-  $release = 'latest',
-  $gpg_key_fingerprint = undef,
+  String              $release             = 'latest',
+  Optional[String[1]] $gpg_key_fingerprint = undef,
+  Boolean             $subscription        = false,
+  Optional[String]    $username            = undef,
+  Optional[String]    $password            = undef,
 ) {
-  $url = "http://download.bareos.org/bareos/release/${release}/"
+  $scheme = 'http://'
```

Comment:
I pretty sure we should change this to https, but probably in another PR.

Reply:
https is available for both .com and .org domains. I'll put that on my todo list for the module and open a PR for it